flax/linen/normalization.py

```diff
@@ -335,6 +343,80 @@ def __call__(self, x):
         self.bias_init, self.scale_init)
 
 
+class RMSNorm(Module):
+  """RMS Layer normalization (https://arxiv.org/abs/1910.07467).
+
+  RMSNorm normalizes the activations of the layer for each given example in a
+  batch independently, rather than across a batch like Batch Normalization.
+  Unlike LayerNorm which re-centers the mean to be 0 and normalizes by the
+  standard deviation of the activations, RMSNorm does not re-center at all
+  and instead normalizes by the root mean square of the activations.
+
+  Example::
+    >>> import jax.numpy as jnp
+    >>> import jax
+    >>> import flax.linen as nn
+    ...
+    >>> x = jax.random.uniform(jax.random.PRNGKey(0), (2, 3))
+    >>> layer = nn.RMSNorm()
+    >>> variables = layer.init(jax.random.PRNGKey(1), x)
+    >>> y = layer.apply(variables, x)
+
+  Attributes:
+    epsilon: A small float added to variance to avoid dividing by zero.
+    dtype: the dtype of the result (default: infer from input and params).
+    param_dtype: the dtype passed to parameter initializers (default: float32).
+    use_bias:  If True, bias (beta) is added.
+    use_scale: If True, multiply by scale (gamma). When the next layer is linear
+      (also e.g. nn.relu), this can be disabled since the scaling will be done
+      by the next layer.
+    bias_init: Initializer for bias, by default, zero.
+    scale_init: Initializer for scale, by default, one.
+    reduction_axes: Axes for computing normalization statistics.
+    feature_axes: Feature axes for learned bias and scaling.
+    axis_name: the axis name used to combine batch statistics from multiple
+      devices. See `jax.pmap` for a description of axis names (default: None).
+      This is only needed if the model is subdivided across devices, i.e. the
+      array being normalized is sharded across devices within a pmap.
+    axis_index_groups: groups of axis indices within that named axis
+      representing subsets of devices to reduce over (default: None). For
+      example, `[[0, 1], [2, 3]]` would independently batch-normalize over
+      the examples on the first two and last two devices. See `jax.lax.psum`
+      for more details.
+  """
+  epsilon: float = 1e-6
+  dtype: Optional[Dtype] = None
+  param_dtype: Dtype = jnp.float32
+  use_bias: bool = True
```

Comment:
sorry I just noticed this - we probably don't want `use_bias` and `bias_init` here since we're never adjusting the offset?